octobot_trading/exchanges/abstract_websocket_exchange.py

```diff
@@ -16,20 +16,64 @@
 #  License along with this library.
 
 import abc
+import asyncio
 
 import octobot_commons.logging as logging
+import octobot_trading.constants
+import octobot_trading.enums
+import octobot_trading.exchange_channel as exchange_channel
+import octobot_trading.exchange_data as exchange_data
 
 
 class AbstractWebsocketExchange:
     __metaclass__ = abc.ABCMeta
 
-    def __init__(self, config, exchange_manager):
+    EXCHANGE_FEEDS = {}
+
+    INIT_REQUIRING_EXCHANGE_FEEDS = set()
+
+    def __init__(self,
+                 config: object,
+                 exchange_manager: object,
+                 api_key: str = None,
+                 api_secret: str = None,
+                 api_password: str = None):
         self.config = config
+        self.api_key = api_key
+        self.api_secret = api_secret
+        self.api_password = api_password
```

Comment:
do we need to store there is clear ? (instead of calling self.exchange_manager.get_exchange_credentials ?

Reply:
You're right, I'll change that